inst/tests/tests.Rraw

```diff
@@ -17273,3 +17273,12 @@ if (test_bit64) {
   test(2164.3, d[, mean(b, na.rm=TRUE), by=a], data.table(a=INT(1,2), V1=c(2.5, 4)))
 }
 
+# Test new feature %notin%, #4152
+test(2165.1, 11 %notin% 1:10, TRUE)
+test(2165.2, "a" %notin% c(), TRUE)
+test(2165.3, "a" %notin% c("a", "b", "c"), FALSE)
+test(2165.4, c(1, 2) %notin% c(1,2,3), c(FALSE, FALSE))
+test(2165.5, "a" %notin% character(), TRUE)
+test(2165.6, "a" %notin% integer(), TRUE)
+test(2165.7, "a" %notin% NULL, TRUE)
```

Comment:
we need tests on `NA` as well. In particular `%in%` _always_ returns `TRUE`/`FALSE`, I guess `%notin%` should work analogously.

It raises a small puzzle -- if `NA %in% 1:5` is `FALSE`, should `NA %notin% 1:5` be `TRUE` or `FALSE`? `TRUE` would be the natural answer to ensure `%in%` and `%notin%` are inverses but I think there could be an argument it should be `FALSE`.

Reply:
Just stumbled upon this as a long time data.table user.

From my perspective, `NA %notin% 1:5` should return `TRUE` to be analogous to `%in%`.
The key observations that leads me to this is that `%in%` treats `NA` as an allowed value in the right-hand-sided vector, in this case `1:5`. To see this, note that `NA %in% c(1:5, NA)` indeed returns `TRUE`.
Having this behaviour in mind, returning `FALSE` is unnatural in the use case you described.

Reply:
I agree with this suggestion and have added tests to reflect this behavior.